### PR TITLE
DOC: Reduce warnings and cleanup redundant c-api documentation

### PR DESCRIPTION
**Pull request by mattip**
There are still many [warnings when building](https://circleci.com/gh/numpy/numpy/2984?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link) documentation, this cleanup removes those that were obvious and moves duplicate documentation of some functions from `user/c-api...` to `reference/c-api...`

The `WARNING: Unexpected section title` are coming from `docutils`, when we have examples in an attribute section. Setting the default `match_titles` parameter in `docutils/parsers/rst/states.py` seems to quiet these:
```
2824,2825c2824
<         self.nested_parse(indented, input_offset=line_offset, node=definition,
<                           match_titles=True)
---
>         self.nested_parse(indented, input_offset=line_offset, node=definition)
```

### CODE REVIEW

File: doc/release/1.13.0-notes.rst [hunk 1]
```diff
@@ -183,10 +183,10 @@ override the behavior of NumPy's ufuncs. This works quite similarly to Python's
 ``__mul__`` and other binary operation routines. See the documentation for a
 more detailed description of the implementation and behavior of this new
 option. The API is provisional, we do not yet guarantee backward compatibility
-as modifications may be made pending feedback. See the NEP_ and
+as modifications may be made pending feedback. See `NEP 13`_  and
 documentation_ for more details.
 
-.. _NEP: https://github.com/numpy/numpy/blob/master/doc/neps/ufunc-overrides.rst
+.. _`NEP 13`: http://www.numpy.org/neps/nep-0013-ufunc-overrides.html
 .. _documentation: https://github.com/charris/numpy/blob/master/doc/source/reference/arrays.classes.rst
```
Comment (charris) [inline doc/release/1.13.0-notes.rst]: Hmm, the documentation points to my repository, seems wrong...

File: doc/release/1.15.0-notes.rst [hunk 2]
```diff
@@ -227,13 +227,6 @@ Changes to ``PyArray_GetDTypeTransferFunction``
 significant performance hit, consider implementing ``copyswapn`` to reflect the
 implementation of ``PyArray_GetStridedCopyFn``.  See `#10898
 <https://github.com/numpy/numpy/pull/10898>`__.
-* Functions ``npy_get_floatstatus_barrier`` and ``npy_clear_floatstatus_barrier``
```
Comment (charris) [inline doc/release/1.15.0-notes.rst]: Duplicate?

File: doc/source/reference/c-api.ufunc.rst [hunk 3]
```diff
@@ -85,12 +85,61 @@ Functions
         Must to an array of length *ntypes* containing
         :c:type:`PyUFuncGenericFunction` items. These items are pointers to
         functions that actually implement the underlying
-        (element-by-element) function :math:`N` times.
+        (element-by-element) function :math:`N` times with the following
+        signature:
+
+        .. c:function:: void loopfunc( \
```
Comment (charris) [inline doc/source/reference/c-api.ufunc.rst]: Is the '\' needed? Just curious.

File: doc/source/reference/c-api.ufunc.rst [hunk 4]
```diff
@@ -114,8 +166,15 @@ Functions
...
+
+        Either :c:data:`PyUFunc_One`, :c:data:`PyUFunc_Zero`,
+        :c:data:`PyUFunc_None`. This specifies what should be returned when
+        an empty array is passed to the reduce method of the ufunc.
+
     :param name:
-        The name for the ufunc.  Specifying a name of 'add' or
+        The name for the ufunc as a ``NULL`` terminated string.
+        Specifying a name of 'add' or
         'multiply' enables a special behavior for  integer-typed
```
Comment (charris) [inline doc/source/reference/c-api.ufunc.rst]: Might justify these lines together for readability here.

File: doc/source/user/c-info.beyond-basics.rst [hunk 5]
```diff
@@ -358,38 +358,6 @@ previously created. Then you call :c:func:`PyUFunc_RegisterLoopForType`
 this function is ``0`` if the process was successful and ``-1`` with
 an error condition set if it was not successful.
 
-.. c:function:: int PyUFunc_RegisterLoopForType( \
```
Comment (charris) [inline doc/source/user/c-info.beyond-basics.rst]: Duplicate?

File: doc/source/user/c-info.ufunc-tutorial.rst [hunk 6]
```diff
@@ -1048,133 +1048,6 @@ The C file is given below.
         #endif
         }
 
-
```
Comment (charris) [inline doc/source/user/c-info.ufunc-tutorial.rst]: Duplicate?

File: numpy/core/_add_newdocs.py [hunk 7]
```diff
@@ -1454,11 +1454,10 @@ def luf(lamdaexpr, *args, **kwargs):
     Values are generated within the half-open interval ``[start, stop)``
     (in other words, the interval including `start` but excluding `stop`).
     For integer arguments the function is equivalent to the Python built-in
-    `range <https://docs.python.org/library/functions.html#func-range>`_ function,
-    but returns an ndarray rather than a list.
+    `range` function, but returns an ndarray rather than a list.
 
     When using a non-integer step, such as 0.1, the results will often not
-    be consistent.  It is better to use ``linspace`` for these cases.
+    be consistent.  It is better to use `numpy.linspace` for these cases.
```
Comment (charris) [inline numpy/core/_add_newdocs.py]: Can we use `np` yet?

Comment (mattip) [reply numpy/core/_add_newdocs.py]: sadly, no. PR #11344 was closed without a fix. Se also PR #11377 which tried to allow the monkeypatching needed to make `np` work.

File: doc/release/1.13.0-notes.rst (hunk 1 above)
Comment (mattip) [reply doc/release/1.13.0-notes.rst]: fixed

File: doc/release/1.15.0-notes.rst (hunk 2 above)
Comment (mattip) [reply doc/release/1.15.0-notes.rst]: yes, it appears above

File: doc/source/reference/c-api.ufunc.rst (hunk 3 above)
Comment (mattip) [reply doc/source/reference/c-api.ufunc.rst]: not clear what you mean. Which "?

Comment (charris) [reply doc/source/reference/c-api.ufunc.rst]: Meant the `\`.

File: doc/source/user/c-info.ufunc-tutorial.rst (hunk 6 above)
Comment (charris) [reply doc/source/user/c-info.ufunc-tutorial.rst]: I was curious about the deletion that starts here.

File: doc/source/reference/c-api.ufunc.rst (hunk 4 above)
Comment (charris) [reply doc/source/reference/c-api.ufunc.rst]: Reflow would be nice.

File: doc/source/reference/c-api.ufunc.rst (hunk 3 above)
Comment (mattip) [reply doc/source/reference/c-api.ufunc.rst]: fixed

File: doc/source/reference/c-api.ufunc.rst (hunk 4 above)
Comment (mattip) [reply doc/source/reference/c-api.ufunc.rst]: fixed

File: doc/source/user/c-info.beyond-basics.rst (hunk 5 above)
Comment (mattip) [reply doc/source/user/c-info.beyond-basics.rst]: Yes

File: doc/source/user/c-info.ufunc-tutorial.rst (hunk 6 above)
Comment (mattip) [reply doc/source/user/c-info.ufunc-tutorial.rst]: Yes, duplicate

**Comment by charris** [comment]
Thanks Matti.

**Pull request merged**